app/download/views.py

```diff
@@ -34,3 +34,31 @@ def download_document(service_id, document_id):
     response.headers['X-Robots-Tag'] = 'noindex, nofollow'
 
     return response
+
+
+@download_blueprint.route('/services/<uuid:service_id>/documents/<uuid:document_id>/check', methods=['GET'])
+def check_document_exists(service_id, document_id):
+    if 'key' not in request.args:
+        return jsonify(error='Missing decryption key'), 400
+
+    try:
+        key = base64_to_bytes(request.args['key'])
+    except ValueError:
+        return jsonify(error='Invalid decryption key'), 400
+
+    try:
+        document_exists = document_store.check_document_exists(service_id, document_id, key)
+    except DocumentStoreError as e:
+        current_app.logger.warning(
+            'Failed to check if document exists: {}'.format(e),
+            extra={
+                'service_id': service_id,
+                'document_id': document_id,
+            }
+        )
+        return jsonify(error=str(e)), 400
+
+    response = make_response({'file_exists': str(document_exists)})
+    response.headers['X-Robots-Tag'] = 'noindex, nofollow'
```

Comment:
We don't have a way of ensuring that a request comes from `document-download-frontend`, and I wasn't sure it was worth adding one just for one route that doesn't return sensitive information, so have just gone with the approach that the download_document route uses to add the `X-Robots-Tag` header instead.

Reply:
i think this endpoint is low risk as doc-dl-frontend is public and un-auth'd anyway. you need info from the frontend url to get the right URL. As in - if you have the service id, document id, and key, then you can either access the doc dl URL or the doc dl api URL.

we should probably have some basic auth, especially if doc dl api starts containing more metadata etc. but out of scope for this story imo though.